Objects/enumobject.c

```diff
@@ -91,31 +103,46 @@ enumerate_vectorcall(PyObject *type, PyObject *const *args,
     PyTypeObject *tp = _PyType_CAST(type);
     Py_ssize_t nargs = PyVectorcall_NARGS(nargsf);
     Py_ssize_t nkwargs = 0;
-    if (nargs == 0) {
-        PyErr_SetString(PyExc_TypeError,
-            "enumerate() missing required argument 'iterable'");
-        return NULL;
-    }
     if (kwnames != NULL) {
         nkwargs = PyTuple_GET_SIZE(kwnames);
     }
 
+    // Manually implement enumerate(iterable, start=...)
     if (nargs + nkwargs == 2) {
         if (nkwargs == 1) {
-            PyObject *kw = PyTuple_GET_ITEM(kwnames, 0);
-            if (!_PyUnicode_EqualToASCIIString(kw, "start")) {
-                PyErr_Format(PyExc_TypeError,
-                    "'%S' is an invalid keyword argument for enumerate()", kw);
+            if (!check_keyword(kwnames, 0, "start")) {
+                return NULL;
+            }
+        } else if (nkwargs == 2) {
+            PyObject *kw0 = PyTuple_GET_ITEM(kwnames, 0);
+            if (_PyUnicode_EqualToASCIIString(kw0, "start")) {
```

Comment:
```suggestion
            if (!check_keyword(kwnames, 0, "start")) {
```

Reply:
No, because check_keyword always sets an error and this branch has a non-error return (in case the user called enumerate(start=..., iterable=...).

Reply:
AH okay undersood!